NickvisionTubeConverter.Shared/Controllers/MainWindowController.cs

```diff
@@ -243,6 +251,43 @@ public void UpdateKeyring(KeyringDialogController controller)
         _keyring = controller.Keyring;
     }
 
+    public void SetTaskbarItem(TaskbarItem? taskbarItem)
```

Comment:
Why not make this a property and not nullable:

```csharp
public TaskbarItem TaskbarItem
{
    set
    {
        _taskbarItem = value;
        DownloadManager.DownloadProgressUpdated += (_, _) => UpdateTaskbar();
        DownloadManager.DownloadCompleted += (_, _) => UpdateTaskbar(true);
        DownloadManager.DownloadStopped += (_, _) => UpdateTaskbar(true);
    }
}
```

Reply:
Idk if [these guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/property?redirectedfrom=MSDN) are relevant for us, but they say:

>  DO NOT provide set-only properties

Also it's nullable because if dbus connection fails TaskbarItem will be null.

Reply:
> DO NOT provide set-only properties

Eh, this guideline I'm okay with breaking...we've been doing it and i've seen most PInvoke/Marshal libraries do the same so it;s fine with me.

> Also it's nullable because if dbus connection fails TaskbarItem will be null.

Ah ok so then the header can be `public TaskbarItem? TaskbarItem` and you can add the null check and return inside the set body.

--

But why is this a public method/property anyway? Isn't all the taskbar code, initializing and all, handled in the controller anyway?

Reply:
> Isn't all the taskbar code, initializing and all, handled in the controller anyway with new Aura?

TaskbarItem creation is platform-specific: `ConnectLinuxAsync` accepts desktop file name, `ConnectWindows` accepts window handle, so TaskbarItem is created in UI and passed to controller.

Reply:
> TaskbarItem creation is platform-specific: ConnectLinuxAsync accepts desktop file name, ConnectWindows accepts window handle, so TaskbarItem is created in UI and passed to controller.

Ah ok got it, then this: `Ah ok so then the header can be public TaskbarItem? TaskbarItem and you can add the null check and return inside the set body.` is fine